greedy_methods/interval_scheduling.py

```diff
@@ -0,0 +1,27 @@
+# Interval Scheduling: https://en.wikipedia.org/wiki/Interval_scheduling
+
+
+def interval_scheduling(intervals):
+    # Sort intervals based on their end time in ascending order
+    sorted_intervals = sorted(intervals, key=lambda x: x[1])
+
+    # Initialize the result list with the first interval (earliest end time)
+    result = [sorted_intervals[0]]
+
+    for interval in sorted_intervals[1:]:
+        # Check if the start time of the current interval is after the end time of the last interval in the result
+        if interval[0] >= result[-1][1]:
+            result.append(interval)
+
+    return result
```

Comment:
```suggestion
    # Sort intervals based on their end time
    sorted_intervals = sorted(intervals, key=lambda x: x[1])

    selected = [sorted_intervals[0]]
    for interval in sorted_intervals[1:]:
        # Add current interval if it doesn't overlap with the last selected interval
        if interval[0] >= result[-1][1]:
            selected.append(interval)

    return selected
```
1) Let's be less verbose in the comments, explain things at a relatively high level and avoid explaining every line of code
2) `result` isn't a very descriptive name, so I suggest changing it to something like `selected`